ap/columnar_util.py

```diff
@@ -745,8 +751,9 @@ def __call__(self, *args, **kwargs):
 class TaskArrayFunction(ArrayFunction):
     """
     Subclass of :py:class:`ArrayFunction` providing an interface to certain task features such as
-    declaring task requirements and dependent or produced shifts, and defining a custom setup
-    function.
+    declaring dependent or produced shifts, task requirements, and defining a custom setup
```

Comment:
AFAIU the `shifts` keyword currently only declares dependent shifts (as in the shifts that affect the result of the array function and for which the column aliases should be applied before the array function is executed). Wouldn't it make sense to also declare produced shifts via a separate keyword?

Reply:
I also thought about that :)
But as far as the ShiftTask mechanism is concerned, dependent and produced shifts don't have a special distinction. For this matter, running with a certain shifts only means that "the output will be different than for 'nominal'".

Reply:
I see, thanks for clarifying. Guess I was just a bit confused by the formulation in the docstring, but when you put it that way it makes sense.